### PR TITLE
reintroduce increment/decrement API on Gauge

### PR DESCRIPTION
**Pull request by jtjeferreira**

### CODE REVIEW

File: kamon-core/src/main/scala/kamon/metric/Gauge.scala [hunk 1]
```diff
@@ -42,6 +62,20 @@ object Gauge {
...
+
+    override def increment(times: Double): Gauge = {
+      _currentValue += times
+      this
+    }
+
+    override def decrement(): Gauge = decrement(1)
+
+    override def decrement(times: Double): Gauge = {
+      _currentValue -= times
```
Comment (ivantopo) [inline kamon-core/src/main/scala/kamon/metric/Gauge.scala]: This should guard against negative values, same as `update` does.

File: kamon-core/src/main/scala/kamon/metric/Gauge.scala [hunk 2]
```diff
@@ -42,6 +62,20 @@ object Gauge {
 
     @volatile private var _currentValue = 0D
 
+    override def increment(): Gauge = increment(1)
+
+    override def increment(times: Double): Gauge = {
```
Comment (ivantopo) [inline kamon-core/src/main/scala/kamon/metric/Gauge.scala]: This should guard against negative increments.

**Comment by jtjeferreira** [comment]
> but we could probably think in the near future whether we should allow negative values in a gauge, although doesn't seem like a priority now.

want me to implement that in this PR?

**Comment by ivantopo** [comment]
Not really. It is not that much about allowing it to happen at the Gauge API level but rather to do the research to ensure that if we allow this to happen here, we will not be breaking any reporters or at least modify them to handle the situation. 

It might be my tired self speaking here, but if we are really releasing tomorrow I would very much like to keep the scope of this change as limited as possible.

**Comment by jtjeferreira** [comment]
> It might be my tired self speaking here, but if we are really releasing tomorrow I would very much like to keep the scope of this change as limited as possible.

Tomorrow? :running_man:

**Comment by ivantopo** [comment]
Yes!

File: kamon-core/src/main/scala/kamon/metric/Gauge.scala (hunk 1 above)
Comment (jtjeferreira) [reply kamon-core/src/main/scala/kamon/metric/Gauge.scala]: @ivantopo should I change `_currentValue` to an `AtomicDouble`?

Comment (ivantopo) [reply kamon-core/src/main/scala/kamon/metric/Gauge.scala]: Is there an AtomicDouble in the JDK?

Comment (jtjeferreira) [reply kamon-core/src/main/scala/kamon/metric/Gauge.scala]: More or less: https://stackoverflow.com/questions/5505460/java-is-there-no-atomicfloat-or-atomicdouble

**Comment by jtjeferreira** [comment]
@ivantopo can you rerun the tests?

**Comment by ivantopo** [comment]
Running again now! I'm not sure why the lifecycle test fails in Travis :(

**Comment by ivantopo** [comment]
Hey @jtjeferreira, I just PRed some tests to your branch: https://github.com/jtjeferreira/Kamon/pull/1

The last 3 tests are failing, something is wrong with increment/decrement, could you please take a look? let me know if you need any help!

**Comment by ivantopo** [comment]
And we are now all green!

**Pull request merged**